pkg/rbd/nodeserver.go

```diff
@@ -38,11 +39,46 @@ type nodeServer struct {
 	*csicommon.DefaultNodeServer
 }
 
+var (
+	pendingVols    = make(map[string]byte)
+	pendingVolsMtx sync.Mutex
+)
+
+func isVolumePending(volID string) bool {
+	pendingVolsMtx.Lock()
+	defer pendingVolsMtx.Unlock()
+
+	_, found := pendingVols[volID]
+	return found
+}
+
+func markPending(volID string) {
+	pendingVolsMtx.Lock()
+	defer pendingVolsMtx.Unlock()
+
+	pendingVols[volID] = 0
+}
+
+func unmarkPending(volID string) {
+	pendingVolsMtx.Lock()
+	defer pendingVolsMtx.Unlock()
+
+	delete(pendingVols, volID)
+}
+
 func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
 	targetPath := req.GetTargetPath()
+	volumeID := req.GetVolumeId()
+
+	if isVolumePending(volumeID) {
+		return nil, fmt.Errorf("rbd: NodePublishVolume for %s is pending", volumeID)
```

Comment:
return a proper error to allow retry 
https://github.com/container-storage-interface/spec/blob/master/spec.md#nodepublishvolume-errors